docs/how-to/how-to-run-dind-with-kata.md

```diff
@@ -0,0 +1,122 @@
+# How to run Docker in Docker with Kata Containers
+
+This document describes the why and how behind running Docker-in-Docker ("dind") in a kata container.
+
+## Requirements
+
+- A working Kata Containers installation
+
+## Install and configure Kata Containers
+
+Follow the [Kata Containers installation guide](../docs/install/README.md) to Install Kata Containers on your Kubernetes cluster.
+
+## Background
+
+Docker in Docker ("DinD") is the colloquial name for the ability to run `docker` from inside a container (whether provided by `docker` or otherwise) pattern for running `docker` from *inside* an existing `docker` container. You can read more about Docker-in-docker at the following links:
+
+- [The original announcement of DinD](https://www.docker.com/blog/docker-can-now-run-within-docker/)
+- [`docker` image DockerHub page](https://hub.docker.com/_/docker/) (this page lists the `-dind` releases)
+
+**NOTE** While in the past kata containers did not contain the [`overlay` kernel module (aka OverlayFS)][overlayfs], it *now does* with the merging of the following pull requests:
+
+- [PR #1384](https://github.com/kata-containers/kata-containers/pull/1384)
+- [PR #716](https://github.com/kata-containers/packaging/pull/716)).
+
+`overlayfs` is the preferred containerization storage driver for linux systems ([Docker in particular](https://docs.docker.com/storage/storagedriver/select-storage-driver)), and as such, using it is often desired over other alternatives.
+
+[overlayfs]: https://www.kernel.org/doc/html/latest/filesystems/overlayfs.html
+
+## Why DinD containers require special measures
+
+Running containers DinD-capable containers with Kata takes more effort than normal due to the fact that `VOLUME`s specified in `Dockerfile`s run by kata containers are given the `kataShared` mount type by default, as it is what is used for the root directory `/`:
+
+```console
+/ # mount
+kataShared on / type virtiofs (rw,relatime,dax)
+```
+
+`kataShared` mount types are powered by [`virtio-fs`][virtio-fs], a marked improvement over `virtio-9p`, thanks to [PR #1016](https://github.com/kata-containers/runtime/pull/1016). While `virtio-fs` is normally an excellent choice, in the case of DinD workloads `virtio-fs` causes an issue -- [it *cannot* be used as a "upper layer" of `overlayfs` without a custom patch](http://lists.katacontainers.io/pipermail/kata-dev/2020-January/001216.html).
+
+As `/var/lib/docker` is a `VOLUME` specified by DinD (i.e. the `docker` images tagged `*-dind`/`*-dind-rootless`), `docker` fill fail to start (or even worse, silently pick a worse storage driver like `vfs`) when started in a Kata Container. Special measures must be taken when running DinD-powered workloads in Kata Containers.
+
+## Workarounds/Solutions
+
+Thanks to various community contributions (see [issue references below](#references)) the following options, with various tradeoffs have been uncovered:
```

Comment:
```suggestion
Thanks to various community contributions (see [issue references below](#references)) the following options, with various trade-offs have been uncovered:
```

tradeoffs -> trade-offs, in order to make the spell-checker happy.

Reply:
We must keep the spell-checker happy
- [ ] "tradeoffs" -> "trade-offs" on line 44